crates/types/src/net/replicated_loglet.rs

```diff
@@ -168,13 +182,95 @@ impl Appended {
 }
 
 // ** GET_TAIL_INFO
-#[derive(Debug, Clone, Serialize, Deserialize)]
+#[derive(Debug, Clone, Serialize, Deserialize, bilrost::Message)]
 pub struct GetSequencerState {
     pub header: CommonRequestHeader,
     pub force_seal_check: bool,
 }
 
-#[derive(Debug, Clone, Serialize, Deserialize)]
+#[derive(Debug, Clone, Serialize, Deserialize, bilrost::Message)]
 pub struct SequencerState {
     pub header: CommonResponseHeader,
 }
+
+mod dto {
+    use super::*;
+
+    #[derive(Debug, Clone, Serialize, Deserialize, bilrost::Oneof, bilrost::Message)]
+    pub enum SequencerStatus {
+        Ok,
```

Comment:
IMHO, using `Ok` here as the default is dangerous because now it also acts as the "empty" state of the status code. Which means we can't tell for sure if it's successful or the value is just unset. 

What I suggest here is to introduce an `Unknown` variant and then move `Ok` to get `bilrost(1)`.

Now, to handle this correct in the `From<SequencerStatus> for super::SequencerStatus` is to map `Unknown` to `super::SequencerStatus::Error(true, "Missing response status code")` or something in that line.